index.d.ts

```diff
@@ -3,4 +3,5 @@ interface Array<T> {
   Aggregate<T>(accum: (memo: T, val: T) => T): T;
   Select<T, U>(xform: (x: T) => U): Array<U>;
   Where<T>(predicate: (X: T) => boolean): Array<T>;
+  Average<number>(): number;
```

Comment:
There is another overload of `Average<T>((x: T) => number): number`. We can merge as-is and leave the issue open. Do you want to merge this as-is?

Reply:
I've covered both cases now in my latest commit.